### PR TITLE
Disable all SwiftLint rules that cause errors for now

### PR DESCRIPTION
**Pull request by thii**
Tested with latest SwiftLint.

### CODE REVIEW

File: tools/generator/src/DTO/Target+LinkerFlags.swift [hunk 1]
```diff
@@ -123,6 +123,7 @@ private func processLinkoptComponent(
             if let `extension` = filePath.path.extension,
                `extension` == "swiftmodule"
             {
+                // swiftlint:disable:next shorthand_operator
                 filePath = filePath + "\(swiftTriple).swiftmodule"
```
Comment (brentleyjones) [inline tools/generator/src/DTO/Target+LinkerFlags.swift:127]: Can we resolve this instead?

File: tools/generator/src/Generator/CreateFilesAndGroups.swift [hunk 2]
```diff
@@ -454,6 +456,7 @@ extension Generator {
             var coreDataContainer: XCVersionGroup?
             let components = fullFilePath.path.components
             for (offset, component) in components.enumerated() {
+                // swiftlint:disable:next shorthand_operator
                 filePath = filePath + component
```
Comment (brentleyjones) [inline tools/generator/src/Generator/CreateFilesAndGroups.swift:460]: Can we resolve this instead?

File: tools/generator/src/Generator/AddBazelDependenciesTarget.swift [hunk 3]
```diff
@@ -18,6 +18,7 @@ extension Generator {
                 files.containsGeneratedFiles)
     }
 
+    // swiftlint:disable:next function_parameter_count
```
Comment (brentleyjones) [inline tools/generator/src/Generator/AddBazelDependenciesTarget.swift:21]: You are now globally disabling this.
```suggestion
```

**Review by brentleyjones — APPROVED** [review]

File: tools/generator/src/DTO/Target+LinkerFlags.swift (hunk 1 above)
Comment (thii) [reply tools/generator/src/DTO/Target+LinkerFlags.swift:127]: It doesn't build

Comment (brentleyjones) [reply tools/generator/src/DTO/Target+LinkerFlags.swift:127]: I wonder if it needs `String()` on the rhs so it knows what the type is, or if we need to define another operator. Anyway, comment is fine for now.

**Pull request merged**